### ISSUE TITLE
Plans 2023: Refactor action handlers and text to be unified and fed into the grid from consuming end

### ISSUE THREAD

**Issue by chriskmnds** [Automattic/wp-calypso#76662]
The [PlanFeatures2023GridActions](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-features-2023-grid/components/actions.tsx) component defines the various action buttons for the grid and the respective text to show. The grid itself accepts an `onUpgradeClick` handler that is fed into`PlanFeatures2023GridActions`. 

The action sequences have been subject to overriding, such as when a modal or some other redirect needs to happen. One case is when the current site is on a free plan, the text for the respective free plan action btn becomes "Manage add-ons" and the action instead redirects to the add-ons page. This is obviously context-dependent (requires knowledge about the current site's plan) and also subject to introducing dependencies that we need to extract (as we are currently doing in https://github.com/Automattic/wp-calypso/pull/76534). The solution is not the cleanest and relies on passing additional specific props into the grid.

I think we should be able to define the text for any respective button along with the action handlers externally (we already do this for the latter) - in effect, `PlanFeatures2023GridActions` becomes a simple presentation layer and doesn't need to know or reason about "logged-in" or "launch" or "signup" or even "logged-out" sections.

For example, roughly thinking, the grid could accept a factory method/hook that unifies the text and click handler into a single object given any plan slug e.g.

```
type PlanAction = {
    callback: () => null;
    text: string;
}

type usePlanAction = ( slug ) => PlanAction
```

**Comment by chriskmnds** [comment]
@southp @jeyip I think this is relevant to the work you are doing with the CTA refactors. Are we kinda close to what is outlined above? :-)

pls clean up this issue if relevant / either assign and move to in-progress or close out?

**Comment by jeyip** [comment]
> I think this is relevant to the work you are doing with the CTA refactors. Are we kinda close to what is outlined above? :-)

Yes! We're moving in this direction. I'll update this issue to reflect our discussion here https://github.com/Automattic/wp-calypso/pull/88694#issuecomment-2035766518.

> either assign and move to in-progress or close out?

I'll leave this issue in "next up" for the time being while we hash out https://github.com/Automattic/wp-calypso/pull/88694#issuecomment-2035766518, which will be addressed in a follow-up
